flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/codegen/calls/ScalarOperators.scala

```diff
@@ -909,7 +909,15 @@ object ScalarOperators {
 
       case (SqlTimeTypeInfo.TIME, TimeIntervalTypeInfo.INTERVAL_MILLIS) =>
         generateOperatorIfNotNull(nullCheck, SqlTimeTypeInfo.TIME, left, right) {
-            (l, r) => s"$l $op ((int) ($r))"
+          (l, r) =>
+            s"((($l % ${MILLIS_PER_DAY} == 0) ? ${MILLIS_PER_DAY} : $l) " +
+              s"+ (java.lang.Math.toIntExact($r % ${MILLIS_PER_DAY} == 0 ? 0 : $r))) " +
+              s"% ${MILLIS_PER_DAY}"
+        }
```

Comment:
this case branch is missing the `$op`. I am assuming that's not expected ? 
(I also saw a `+` sign on line 914 which shouldn't be there) 

is this branch tested?